core/src/main/scala/cats/syntax/monadError.scala

```diff
@@ -31,6 +38,11 @@ final class MonadErrorOps[F[_], E, A](private val fa: F[A]) extends AnyVal {
     F.adaptError(fa)(pf)
 }
 
+final class MonadErrorOps0[F[_], E, A](private val fa: F[A]) extends AnyVal {
```

Comment:
I don't think any of the bin compat traits are needed. we can just add this method to the existing `MonadErrorOps`.
Also let's have a doctest. 
But more fundamentally would you help me understand the incentive for this? Because I think there might be something else missing. 
What if we have the following syntax on `MonadError`
`def foldF[B]((e: E) => F[B])((a: A) => F[B]) : F[B]`
and then you can do 
`me.foldF(onError)(onSuccess) *> me`
and if that's too much typing 
we can add a `foldTapE`
Would that work better for you than involving Either?